internal/controller/management/serviceaccount/serviceaccounts.go

```diff
@@ -0,0 +1,229 @@
+
+	// Iterate over each project and create RoleBinding
+	for _, project := range projectList.Items {
+
+		roleBinding := &rbacv1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      roleBindingName,
+				Namespace: project.Name, // Project name is the namespace name
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "ClusterRole",
+				Name:     "kargo-controller-secrets-readonly",
+			},
+			Subjects: []rbacv1.Subject{
+				{
+					Kind:      "ServiceAccount",
+					Name:      sa.Name,
+					Namespace: r.cfg.KargoNamespace,
+				},
+			},
+		}
+
+		// Create or update the RoleBinding		
+		if err := r.Client.Get(ctx, client.ObjectKey{Name: roleBindingName, Namespace: project.Name}, roleBinding); err != nil {
+			if kubeerr.IsNotFound(err) {
+				// Create RoleBinding
+				if err := r.Client.Create(ctx, roleBinding); err != nil {
+					return fmt.Errorf("error creating RoleBinding %q for ServiceAccount %q in project namespace %q: %w",
+						roleBinding.Name, sa.Name, project.Name, err)
+				}
+				logger.Info("Created RoleBinding for ServiceAccount", "roleBinding", roleBindingName, "namespace", project.Name)
+			} else {
+				return fmt.Errorf("error retrieving RoleBinding %q in namespace %q: %w", roleBindingName, project.Name, err)
+			}
+		} else {
+			// Update RoleBinding
+			if err := r.Client.Update(ctx, roleBinding); err != nil {
+				return fmt.Errorf("error updating RoleBinding %q in project namespace %q: %w", roleBinding.Name, project.Name, err)
+			}
+			logger.Debug("Updated RoleBinding for ServiceAccount", "roleBinding", roleBindingName, "namespace", project.Name)
+		}
+	}
+	logger.Info("Completed creating RoleBindings for all projects")
+	return nil
+}
+
+// deleteRoleBindingsForAllProjects deletes RoleBindings for all projects
+func (r *ServiceAccountReconciler) deleteRoleBindingsForAllProjects(ctx context.Context, sa *corev1.ServiceAccount) error {
+
+	roleBindingName := fmt.Sprintf("%s-readonly-secrets", sa.Name)
+
+	logger := logging.LoggerFromContext(ctx).WithValues(
+		"serviceaccount", sa.Name,
+		"roleBinding", roleBindingName,
+	)
+	logger.Info("Starting to delete RoleBindings for all projects")
+
+	projectList := &kargoapi.ProjectList{}
+	if err := r.Client.List(ctx, projectList); err != nil {
+		return fmt.Errorf("error listing projects: %w", err)
+	}
+
+	for _, project := range projectList.Items {
+		
+		roleBinding := &rbacv1.RoleBinding{}
+
+		err := r.Client.Get(ctx, client.ObjectKey{Name: roleBindingName, Namespace: project.Namespace}, roleBinding)
+		if err != nil {
+			if kubeerr.IsNotFound(err) {
+				logger.Debug("RoleBinding not found, nothing to remove", "roleBinding", roleBindingName, "namespace", project.Namespace)
+				continue // Skip to the next project if not found
+			}
+			return fmt.Errorf("error retrieving RoleBinding %q in namespace %q: %w", roleBindingName, project.Namespace, err)
+		}
+
+		// Delete the RoleBinding if it exists
```

Comment:
Nit on readability. We _already_ dealt with the possibility of the RB not existing, so don't treat this as "if"

```suggestion
		// Delete the RoleBinding
```